pkg/server/backuprestoreserver.go

```diff
@@ -362,16 +362,12 @@ func (b *BackupRestoreServer) runEtcdProbeLoopWithSnapshotter(ctx context.Contex
 			// the delta snapshot memory limit), after which a full snapshot
 			// is taken and the regular snapshot schedule comes into effect.
 
-			// TODO: write code to find out if prev full snapshot is older than it is
-			// supposed to be, according to the given cron schedule, instead of the
-			// hard-coded "24 hours" full snapshot interval
-
 			// Temporary fix for missing alternate full snapshots for Gardener shoots
 			// with hibernation schedule set: change value from 24 ot 23.5 to
 			// accommodate for slight pod spin-up delays on shoot wake-up
 			const recentFullSnapshotPeriodInHours = 23.5
 			initialDeltaSnapshotTaken = false
-			if ssr.PrevFullSnapshot != nil && !ssr.PrevFullSnapshot.IsFinal && time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() <= recentFullSnapshotPeriodInHours {
+			if ssr.PrevFullSnapshot != nil && !ssr.PrevFullSnapshot.IsFinal && !ssr.IsScheduledFullSnapshotMissed() {
```

Comment:
Using `IsScheduledFullSnapshotMissed` is not the right thing to do here. The earlier condition was sufficient. You should club the new condition with `if !initialDeltaSnapshotTaken` and OR it with this.

Reply:
> The earlier condition was sufficient

what makes you think that? for that condition this issue of alerts happened.

Reply:
> You should club the new condition with if !initialDeltaSnapshotTaken and OR it with this.

I don't think so, else we end up taking delta snapshot first then decide whether to take a full snapshot or not.

Reply:
I have address it now, can you PTAL